pkg/dashboard/dashboard.go

```diff
@@ -151,13 +151,16 @@ func GetRouter(c conf.Configuration, auditPath string, port int, basePath string
 		w.Write(favicon)
 	})
 	router.HandleFunc("/results.json", func(w http.ResponseWriter, r *http.Request) {
-		k, err := kube.CreateResourceProvider(auditPath)
-		if err != nil {
-			logrus.Errorf("Error fetching Kubernetes resources %v", err)
-			http.Error(w, "Error fetching Kubernetes resources", http.StatusInternalServerError)
-			return
+		if auditData == nil {
```

Comment:
We still need to handle the case where `auditData != nil`

I think the best thing to do is move `RunAudit` out of `JSONHandler` up to here, then pass the audit to `JSONHandler`.

Then you can move the call to `JSONHandler` outside of this `if` block - so
```go
if auditData == nil {
  k, err := kube.CreateResourceProvider(auditPath)
  // handle err
  auditData, err = validator.RunAudit(c, k)
  // handle err
}
JSONHandler(w, r, c, auditData)
```